tests/test_MatchPessimisticB.py

```diff
@@ -0,0 +1,215 @@
+
+    def setUp(self):
+
+        self.config = measAstrom.MatchPessimisticBTask.ConfigClass()
+        self.MatchPessimisticB = measAstrom.MatchPessimisticBTask(config=self.config)
+
+        metadata = dafBase.PropertySet()
+        metadata.set("RADECSYS", "FK5")
+        metadata.set("EQUINOX", 2000.0)
+        metadata.set("CTYPE1", "RA---TAN")
+        metadata.set("CTYPE2", "DEC--TAN")
+        metadata.set("CUNIT1", "deg")
+        metadata.set("CUNIT2", "deg")
+        metadata.set("CRVAL1", 36.930640)
+        metadata.set("CRVAL2", -4.939560)
+        metadata.set("CRPIX1", 792.4)
+        metadata.set("CRPIX2", 560.7)
+        metadata.set("CD1_1", -5.17e-05)
+        metadata.set("CD1_2", 0.0)
+        metadata.set("CD2_2", 5.17e-05)
+        metadata.set("CD2_1", 0.0)
+        self.wcs = afwImage.makeWcs(metadata)
+        self.distortedWcs = self.wcs
+
+        self.filename = os.path.join(os.path.dirname(__file__), "cat.xy.fits")
+        self.tolArcsec = .4
+        self.tolPixel = .1
+
+    def tearDown(self):
+        del self.config
+        del self.MatchPessimisticB
+        del self.wcs
+        del self.distortedWcs
+
+    def testLinearXDistort(self):
+        self.singleTestInstance(self.filename, distort.linearXDistort)
+
+    def testLinearYDistort(self):
+        self.singleTestInstance(self.filename, distort.linearYDistort)
+
+    def testQuadraticDistort(self):
+        self.singleTestInstance(self.filename, distort.quadraticDistort)
+
+    def testLargeDistortion(self):
+        # This transform is about as extreme as I can get:
+        # using 0.0005 in the last value appears to produce numerical issues.
+        # It produces a maximum deviation of 459 pixels, which should be sufficient.
+        pixelsToTanPixels = afwGeom.RadialXYTransform([0.0, 1.1, 0.0004])
+        self.distortedWcs = afwImage.DistortedTanWcs(self.wcs, pixelsToTanPixels)
+
+        def applyDistortion(src):
+            out = src.table.copyRecord(src)
+            out.set(out.table.getCentroidKey(),
+                    pixelsToTanPixels.reverseTransform(src.getCentroid()))
+            return out
+
+        self.singleTestInstance(self.filename, applyDistortion)
+
+    def singleTestInstance(self, filename, distortFunc, doPlot=False):
+        sourceCat = self.loadSourceCatalog(self.filename)
+        refCat = self.computePosRefCatalog(sourceCat)
+        distortedCat = distort.distortList(sourceCat, distortFunc)
+
+        if doPlot:
+            import matplotlib.pyplot as plt
+            undistorted = [self.wcs.skyToPixel(self.distortedWcs.pixelToSky(ss.getCentroid())) for
+                           ss in distortedCat]
+            refs = [self.wcs.skyToPixel(ss.getCoord()) for ss in refCat]
+
+            def plot(catalog, symbol):
+                plt.plot([ss.getX() for ss in catalog], [ss.getY() for ss in catalog], symbol)
+
+            # plot(sourceCat, 'k+') # Original positions: black +
```

Comment:
Folks tend to get shirty about commented-out code. If you really don't find this useful can you delete it? Otherwise please explain why it's commented out and when it can be removed or re-enabled.